tutorials/basic-topics/sorting/insertion-sort.md

```diff
@@ -11,4 +11,222 @@ keywords:
+
+First of all, let us understand how an sorted `array` (If you don't what an array is. Please refer [here](/tutorials/basic-topics/arrays.md)) looks like. Understanding an sorted array will give us the overview of the insertion sort. An sorted array is given below.
+
+Example: [1,2,3,4,5,6,7]
+
+In a sorted `array` all the elements to the left any element are smaller than that. That, means to make the `array` sorted we have to place an element in such a position, so that every element to the left are smaller than that. That is pretty much what we do in a insertion sort.
+
+## Algorithm
+
+The following are the steps that we are going to follow:
+
+1. Pick an element from the array, and store it in a variable `nums`.
+2. Now, start comparing the `nums` with all the elements to the left of it.
+3. If the `nums` is less than the current element shift the current element to right until you find an element smaller than `nums`.
+4. Place the `nums` in current position.
+5. Repeat all the above steps until the `array` is sorted.
+
+If we apply the above steps for the array `[3,2,5,10,9]`, it would be:
+* First of all, we will pick the first element which is `3` in our case.
+* Now, we will compare with all the elements left to it, in this case there is nothing left to `3` so, we will do nothing.
+* Now, pick the next element which is `2` and start comparing with all the elements left to it.
+* First element left to `2` is `3` , which is greater than selected element `2`. So, we will move `3` to right. If the element left to selected element is less than selected element then, our selected element have reached it's correct position so, we will place selected element there only.
+* Now, there is nothing no more element further left to our selected element `2`. So, we will place `2` at the beginning of array.
+* We will proceed with the same way for all the elements. Then, at last we will get our array as sorted.
+
+### Problem Link
+
+https://leetcode.com/problems/maximum-product-of-two-elements-in-an-array/
+
+### Problem Statement
+
+Given the array of integers nums, you will choose two different indices i and j of that array. Return the maximum value of `(nums[i]-1)*(nums[j]-1)`.
+
+**Example 1:**
+
+```
+Input: nums = [3,4,5,2]
+Output: 12 
+Explanation: If you choose the indices i=1 and j=2 (indexed from 0), you will get the maximum value, that is, (nums[1]-1)*(nums[2]-1) = (4-1)*(5-1) = 3*4 = 12. 
+```
+
+**Example 2:**
+
+```
+Input: nums = [1,5,4,5]
+Output: 16
+Explanation: Choosing the indices i=1 and j=3 (indexed from 0), you will get the maximum value of (5-1)*(5-1) = 16.
+```
+
+**Example 3:**
+
+```
+Input: nums = [3,7]
+Output: 12
+```
+
+**Constraints:**
+
+* `2 <= nums.length <= 500`
+* `1 <= nums[i] <= 10^3`
+
+## Approach: Sorting using Insertion Sort
+
+The problem wants us to find the maximum value of `(nums[i] - 1) * (nums[j] - 1)`, where `i` and `j` are two different indices of the given array.
```

Comment:
Instead of backtick, use LaTeX for this **(nums[i] - 1) * (nums[j] - 1)**

Change it wherever you've mentioned the same.